src/tests/ftest/aggregation/dfuse_space_check.py

```diff
@@ -0,0 +1,164 @@
+
+          Args:
+            display (bool): boolean to display output of free space.
+
+          Returns:
+            free_space_nvme (int): Free space available in nvme.
+        """
+        free_space_nvme = self.pool.get_pool_free_space("nvme")
+        if display:
+            self.log.info("Free nvme space: %s", free_space_nvme)
+
+        return free_space_nvme
+
+    def check_aggregation(self):
+        """Check Aggregation for 240 secs max, else fail the test."""
+        counter = 1
+        while self.get_nvme_free_space() != self.space_before:
+            # try to wait for 4 x 60 secs for aggregation to be completed or
+            # else exit the test with a failure.
+            if counter > 4:
+                self.log.info("Free space when test terminated: %s",
+                              self.get_nvme_free_space())
+                self.fail("Aggregation did not complete as expected")
+            time.sleep(60)
+            counter += 1
+
+    def write_multiple_files(self):
+        """Write multiple files
+
+          Returns:
+            file_count (int): Total number of files created before
+                              going out of space.
+        """
+        file_count = 0
+        while self.get_nvme_free_space(False) >= self.block_size:
+            file_loc = str(self.dfuse.mount_dir.value +
+                           "/largefile_{}.txt".format(file_count))
+            write_dd_cmd = u"dd if=/dev/zero of={} bs={} count=1".format(
+                file_loc, self.block_size)
+            if 0 in self.execute_cmd(write_dd_cmd, False, False):
+                file_count += 1
+
+        return file_count
+
+    def test_dfusespacecheck(self):
+        """Jira ID: DAOS-3777.
+
+        Test Description:
+            Purpose of this test is to mount dfuse and verify aggregation
+            to return space when pool is filled with once large file and
+            once with small files.
+        Use cases:
+            Create Pool
+            Create Posix container
+            Mount dfuse
+            Create largefile.txt and write to it until pool is out of space.
+            Remove largefile.txt and wait for aggregation to return all the
+            space back.
+            Now create many small files until pool is out of space again and
+            store the number of files created.
+            Remove all the small files created and wait for aggregation to
+            return all the space back.
+            Now create small files again until pool is out of space and check,
+            whether out of space happens at the same file count as before.
+        :avocado: tags=all,hw,daosio,small,full_regression,dfusespacecheck
+        """
+        # get test params for cont and pool count
+        self.block_size = self.params.get("block_size",
+                                          '/run/dfusespacecheck/*')
+
+        # Create a pool, container and start dfuse.
+        self.create_pool()
```

Comment:
You can call self.add_pool() instead. Then you don't need to use create_pool() in ior_test_base.py.